### PR TITLE
Nuzhy/CSIT-1457/freshchat integration

### PR DESCRIPTION
**Pull request by Nuzhy-Deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

**Review by prince-deriv — DISMISSED** [review]

File: src/javascript/_common/analytics.js [hunk 1]
```diff
@@ -1,11 +1,43 @@
...
+const Cookies = require('js-cookie');
+const { getCountry } = require('./utility');
+const LocalStore = require('./storage').LocalStore;
+const Language = require('./language');
+const { getAppId } = require('../config');
 
 const Analytics = (() => {
-    const init = () => {
+    const init = async () => {
+        const account_type = LocalStore?.get('active_loginid')
```
Comment (henry-deriv) [inline src/javascript/_common/analytics.js:10]: Screenshot 2024-11-06 at 4 06 28 PM

you dont need to do this.

client.accounts has something called broker for each account object

File: src/javascript/app/base/header.js [hunk 2]
```diff
@@ -649,6 +656,10 @@ const Header = (() => {
     };
 
     const logoutOnClick = async () => {
+        window.fcWidget?.user.clear().then(
+            () => window.fcWidget.destroy(),
```
Comment (henry-deriv) [inline src/javascript/app/base/header.js:660]: freshchat doesnt work when you are logged out? Or just clearing previous user data?

File: src/javascript/app/hooks/use-freshChat.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
```
Comment (henry-deriv) [inline src/javascript/app/hooks/use-freshChat.js]: just a small nit on the file name useFreshchat or something

File: src/javascript/_common/analytics.js (hunk 1 above)
Comment (Nuzhy-Deriv) [inline src/javascript/_common/analytics.js]: updated. please review

File: src/javascript/app/base/header.js (hunk 2 above)
Comment (Nuzhy-Deriv) [reply src/javascript/app/base/header.js:660]: when logged out we are clearing the chat and destroying freshchat widget. freshchat will get reinitialized without the re-route.

File: src/javascript/app/hooks/use-freshChat.js (hunk 3 above)
Comment (Nuzhy-Deriv) [reply src/javascript/app/hooks/use-freshChat.js]: renamed to useFreshChat

**Review by henry-deriv — APPROVED** [review]

File: src/javascript/_common/analytics.js [hunk 4]
```diff
@@ -1,11 +1,43 @@
...
         if (process.env.RUDDERSTACK_KEY && process.env.GROWTHBOOK_CLIENT_KEY) {
             DerivAnalytics.Analytics.initialise({
-                growthbookKey : process.env.GROWTHBOOK_CLIENT_KEY, // optional key to enable A/B tests
-                rudderstackKey: process.env.RUDDERSTACK_KEY,
+                growthbookKey    : process.env.GROWTHBOOK_CLIENT_KEY, // optional key to enable A/B tests
+                rudderstackKey   : process.env.RUDDERSTACK_KEY,
+                growthbookOptions: {
+                    attributes: {
+                        loggedIn       : !!Cookies.get('clients_information'),
+                        account_type   : active_account?.account_type || 'unlogged',
```
Comment (henry-deriv) [inline src/javascript/_common/analytics.js:27]: its active_account?.broker

Comment (henry-deriv) [inline src/javascript/_common/analytics.js:27]:
```suggestion
                        account_type   : active_account?.broker || 'unlogged',
```

**Review by akmal-deriv — APPROVED** [review]

**Review by ahmadtaimoor-deriv — DISMISSED** [review]

**Review by prince-deriv — APPROVED** [review]

**Pull request merged**